### DISCUSSION TITLE
Project titles and document file names should be decoupled

### DISCUSSION THREAD

**Discussion by vsanchez** [Zettlr/Zettlr#2609] [Ideas]
As far as I can see, the project title as defined in the Properties popup becomes not only the document title but also the name of the file created. I think this is a problem because a document title can easily have a form not compatible with the file system, or simply be too long. I much prefer having the document name be the same as the directory name (as it works by default) which always will be a valid file name and you can always use a directory name that makes sense to describe the document and limit the "Project title" to define the actual title of the document.

The other option would to take the "Title" from the yaml header of the first file (which seems to be used for everything else) and let then the title be the file name of the document..

¿Does anybody requires it to behave in the current form? 

 I personally would prefer not to have to specify neither in the properties:

1. The document file name is the same as the directory name with the proper extension.
2. The title of the whole document comes form the first file of the project.

**Comment by nathanlesage** [comment]
> because a document title can easily have a form not compatible with the file system

Before using the title as a filename, we're running it through a sanitizer to make sure that that's not the case.

Other than that, it seems to me difficult to discern to which parts you refer, since it seems a bit convoluted. Do you refer only to projects, or to any file? It would be great if you could be a tad more specific in this regard, this helps a lot! Thanks already for clarification!

**Reply by vsanchez** [reply]
Only with projects. I did notice the use of a sanitizer but it was still awkward. In my case, the proposed title started,  for example with the opening question mark used in Spanish (¿) , no problem for me but I rather not use it as part of the file name. And the file name was very long. 

Perhaps the question is why to treat the "title" differently from all the other variables in the document. I am not sure what are you doing with the YAML section of all the files when you concatenate them. But it does use most of the variables defined in the first file (at least) to fill up the template. Why treat differently the title? In fact, for a large project, you may want to have the first file (01-settings.md) containing the configuration for the whole document (margins, paper sizes, etc ) and not part of the actual chapters or sections.

**Reply by nathanlesage** [reply]
The main reason why we did that is because Pandoc can sometimes be a little bit picky when it comes to merging variables, so this way it's more explicit. I can't think of a better way to do this :/ 

> In fact, for a large project, you may want to have the first file (01-settings.md) containing the configuration for the whole document (margins, paper sizes, etc ) and not part of the actual chapters or sections.

There is an issue open which I'll address sometime later which will introduce custom defaults files solely for the purpose of doing precisely that, since some options cannot be set using YAML frontmatters.

**Reply by vsanchez** [reply]
Yes, I see now that a "default" file can be used to define a Metadata file. I can always hardcode the title in the particular template of a project, to avoid the file naming issues.
